### ISSUE TITLE
osx builds failing because BasicTex download fails

### ISSUE THREAD

**Issue by jayhesselberth** [travis-ci/travis-ci#6702]
``` bash
$ wget http://mirror.ctan.org/systems/mac/mactex/BasicTeX.pkg -O "/tmp/BasicTeX.pkg"

...

Connecting to ctan.mirrors.hoobly.com|66.160.172.98|:80... connected.
HTTP request sent, awaiting response... 403 Forbidden
2016-10-10 15:13:32 ERROR 403: Forbidden.

The command "wget http://mirror.ctan.org/systems/mac/mactex/BasicTeX.pkg -O "/tmp/BasicTeX.pkg"" failed and exited with 8 during .

Your build has been stopped.

/Users/travis/build.sh: line 109: shell_session_update: command not found
```

**Comment by jayhesselberth** [comment]
works now, fixed? or intermittent?

**Comment by krivit** [comment]
This exact error just happened to my build. (https://travis-ci.org/statnet/network/jobs/246375607)

**Comment by aaronrudkin** [comment]
@jayhesselberth 
We're getting these errors intermittently on a project I'm on. What seems to happen is that, as above, the CTAN mirroring system redirects Travis to a bad mirror, which throws an error, and rather than recovering gracefully by trying a different mirror, it fails instantly. I can replicate it just by running the wget line on a local machine repeatedly and observe that it fails a lot. Obviously it's CTAN's responsibility to deal with server downtime, but Travis should have graceful behaviour.

One presumes that something like would produce more graceful behaviour on Travis' side.:
`wget --retry-on-http-error=403 --waitretry=3 --tries=5 http://mirror.ctan.org/systems/mac/mactex/BasicTeX.pkg -O "/tmp/BasicTeX.pkg"`

I wouldn't begin to know where in Travis to submit this change, but hopefully someone can integrate this pretty quickly?

**Comment by jayhesselberth** [comment]
I have found this is intermittent. Try rerunning the build a few times and it should resolve itself.

**Comment by aaronrudkin** [comment]
Yes, I noted that above. But for us it fails a pretty lengthy amount of time into the build process and it'd be nice if we had more graceful failure behaviour when MacTeX gives us a bum mirror.

The solution is to change line 453 of travis-build/lib/travis/build/script/r.rb to read as above.

**Comment by graemeblair** [comment]
+1 

This breaks our build quite frequently, adding time in our workflow to check and restart the builds and leaves the status "errored" when we miss it. Would much appreciate the fix suggested by @aaronrudkin

**Comment by BanzaiMan** [comment]
For those who are interested, the fix would go here: https://github.com/travis-ci/travis-build/blob/26ec03cbd148873cab6eaa1f155022485fcf06d4/lib/travis/build/script/r.rb#L453

**Comment by maxheld83** [comment]
this is just to confirm that the problem persists, if intermittently.

Can I suggest this be reopened, since it is not fixed @jayhesselberth?

as @aaronrudkin noted, the root cause may lie with CTAN, but it would be nice to have a workaround as suggested in the above.

I'm guessing this will affect *all* `os: osx` `language: r` builds, because that environment always needs R; therefore paging @craigcitro @hadley and @jimhester.
